pkg/dashboard/resource/project.go

```diff
@@ -134,23 +257,116 @@ func (pr *projectResource) Create(request *http.Request) (id string, attributes
+		}
+	}
+
+	pr.Logger.Debug("Importing project functions")
+
+	functionCreateChan := make(chan restful.Attributes, len(projectImportInfoInstance.Functions))
+
+	var failedFunctions []restful.Attributes
+	for functionName, functionImport := range projectImportInfoInstance.Functions {
+		go func(functionName string, functionImport *functionInfo) {
+			functionImport.Meta.Namespace = projectImportInfoInstance.Project.Meta.Namespace
+			if functionImport.Meta.Labels == nil {
+				functionImport.Meta.Labels = map[string]string{}
+			}
+			functionImport.Meta.Labels["nuclio.io/project-name"] = projectImportInfoInstance.Project.Meta.Name
+
+			err = pr.importFunction(functionImport, authConfig)
+			if err != nil {
+				pr.Logger.WarnWith("Failed posting function", "functionName", functionName, "err", err)
+				functionCreateChan <- restful.Attributes{
+					"function": functionName,
+					"error":    err.Error(),
+				}
+			}
+			functionCreateChan <- nil
+		}(functionName, functionImport)
+	}
+
+	for range projectImportInfoInstance.Functions {
+		createError := <-functionCreateChan
+		if createError != nil {
+			failedFunctions = append(failedFunctions, createError)
+		}
+	}
+	close(functionCreateChan)
+
+	attributes = restful.Attributes{
+		"createdFunctionAmount": len(projectImportInfoInstance.Functions) - len(failedFunctions),
+		"failedFunctionsAmount": len(failedFunctions),
+		"failedFunctions":       failedFunctions,
+	}
+
+	return
+}
+
+func (pr *projectResource) importFunction(functionInfo *functionInfo, authConfig *platform.AuthConfig) error {
+	functions, err := pr.getPlatform().GetFunctions(&platform.GetFunctionsOptions{
+		Name:      functionInfo.Meta.Name,
+		Namespace: functionInfo.Meta.Namespace,
+	})
+	if err != nil {
+		return errors.New("Failed to get functions")
+	}
+	if len(functions) > 0 {
+		return errors.New("Function name already exists")
+	}
+
+	// validation finished successfully - store and deploy the given function
+	return functionResourceInstance.storeAndDeployFunction(functionInfo, authConfig, false)
+}
+
+func (pr *projectResource) createAndWaitForProjectCreation(projectInfoInstance *projectInfo) error {
+
+	// create a project config
+	projectConfig := platform.ProjectConfig{
+		Meta: *projectInfoInstance.Meta,
+		Spec: *projectInfoInstance.Spec,
+	}
+
+	// create a project
+	newProject, err := platform.NewAbstractProject(pr.Logger, pr.getPlatform(), projectConfig)
+	if err != nil {
+		return nuclio.WrapErrInternalServerError(err)
+	}
+
+	// just deploy. the status is async through polling
+	err = pr.getPlatform().CreateProject(&platform.CreateProjectOptions{
+		ProjectConfig: *newProject.GetConfig(),
+	})
+	if err != nil {
+		return nuclio.WrapErrInternalServerError(err)
+	}
+
+	timeout := 30 * time.Second
+	retryInterval := 1 * time.Second
+	err = common.RetryUntilSuccessful(timeout, retryInterval, func() bool {
+		pr.Logger.DebugWith("Trying to get projects", "timeout", timeout, "retryInterval", retryInterval)
```

Comment:
its not just projects, its a particular project your waiting on, add the project name (from meta)